jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/ScanningAppProvider.java

```diff
@@ -58,7 +55,6 @@ public abstract class ScanningAppProvider extends ContainerLifeCycle implements
     private int _scanInterval = 10;
     private Scanner _scanner;
     private boolean _useRealPaths;
-    private String _environmentName;
     private boolean _deferInitialScan = false;
 
     private final Scanner.DiscreteListener _scannerListener = new Scanner.DiscreteListener()
```

Comment:
Make this a BulkListener so the list can be sorted and XML vs WAR files better resolved.
I also don't think we need the distinction below of added, changed or removed.  For any given path we simply remove any existing App for that, then if the path exists we deploy it.